### PR TITLE
docs: update `CONTRIBUTING.md`

### PR DESCRIPTION
**Pull request by magnus-madsen**

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,45 +1,57 @@
...
-    We try to ensure that multiple people are not working on the same issue -- at least not without being aware of it.
+2. Express interest in the issue on GitHub.
+    - We try to ensure that multiple people are not working on the same issue.
  
 3. We discuss the issue and how best to fix it.
-
-    The idea is to come up with an implementation plan and to hash out any design choices.
+    - The idea is to come up with an implementation plan and to hash out any design choices.
 
 4. You fork the Flix repository to your own account on GitHub.
```
Comment (mlutze) [inline docs/CONTRIBUTING.md:18]: We got rid of "you" in the other steps

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -1,45 +1,57 @@
...
 
-You affirm this by adding yourself to AUTHORS.md.
+We are interested in correct, well-documented, and well-tested code. 
 
+The argument that "the code works" is not sufficient. It must be maintainable. It must follow our style.
 
+We refer to [STYLE.md](./STYLE.md) for more information on code style.
+- As an example of high-quality code, we refer to `FastBoolUnification.scala`.
 
+Getting your code merged into Flix is a privilege, not a right.
```
Comment (mlutze) [inline docs/CONTRIBUTING.md:57]: Can we tone this down slightly? Maybe just "We expect all PRs to follow the established style and to be maintainable..."

The way it is now feels overly combative.

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -1,45 +1,57 @@
...
 
 4. You fork the Flix repository to your own account on GitHub.
 
-5. You clone this repository to your local machine.
+5. You clone the forked repository to your own machine.
 
 6. You make the required changes, add test cases, documentation, and so forth.
 
 7. You push the local changes to your GitHub repository and create a new pull request (PR).
-   - See [Pull Requests](PULLREQUEST.md) for information on formatting
```
Comment (mlutze) [inline docs/CONTRIBUTING.md:30]: Does GitHub treat that file specially? E.g. displaying it when someone opens a PR the first time?

Comment (magnus-madsen) [reply docs/CONTRIBUTING.md:30]: No, there is no support for that.

**Comment by magnus-madsen** [comment]
All good?

**Review by mlutze — APPROVED** [review]

**Pull request merged**